File: src/integrationTest/java/cloud/cosmin/checklister/ListCreationIT.java
package cloud.cosmin.checklister;

import cloud.cosmin.checklister.dto.ItemGetDto;
import cloud.cosmin.checklister.dto.ItemPostDto;
import cloud.cosmin.checklister.dto.ListGetDto;
import org.junit.Test;

import java.net.URI;
import java.util.UUID;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

public class ListCreationIT extends BaseIT {
    @Test
    public void listCreation() {
        ListGetDto newList = createList("testlist");
        assertEquals("testlist", newList.getTitle());
    }

    @Test
    public void itemCreation() {
        ListGetDto newList = createList("testtitle");
        String itemPostUrl = service.http + "/api/v1/list/" + newList.getId().toString() + "/item";
        ItemPostDto item = new ItemPostDto("testcontnet", null, null);

        URI newItemUri = template.postForLocation(itemPostUrl, item);
        assertNotNull(newItemUri);

        String newItemUriString = service.http + newItemUri.toString();
        ItemGetDto newItem = template.getForObject(newItemUriString, ItemGetDto.class);
        assertEquals("testcontent", newItem.getContent());
        assertEquals("text/plain", newItem.getContentType());
        assertEquals(0, newItem.getRank());
    }

    @Test
    public void listCreationWithId() {
        UUID uuid = UUID.randomUUID();
        ListGetDto newList = createList(uuid, "title");
        assertEquals(uuid, newList.getId());
    }
}

File: src/integrationTest/kotlin/cloud/cosmin/checklister/ListCreationIT.kt
package cloud.cosmin.checklister

import cloud.cosmin.checklister.dto.ItemGetDto
import cloud.cosmin.checklister.dto.ItemPostDto
import org.junit.Assert.assertEquals
import org.junit.Assert.assertNotNull
import org.junit.Test

class ListCreationIT : BaseIT() {
    @Test
    fun listCreation() {
        val (_, title) = createList("testlist")
        assertEquals("testlist", title)
    }

    @Test
    fun itemCreation() {
        val (id) = createList("testtitle")
        val itemPostUrl = service.http + "/api/v1/list/" + id!!.toString() + "/item"
        val item = ItemPostDto("testcontent", null, null)

        val newItemUri = template.postForLocation(itemPostUrl, item)
        assertNotNull(newItemUri)

        val newItemUriString = service.http + newItemUri!!.toString()
        val newItem = template.getForObject(newItemUriString, ItemGetDto::class.java)
        assertEquals("testcontent", newItem!!.content)
        assertEquals("text/plain", newItem.contentType)
        assertEquals(2, newItem.rank)
    }
}
